### PR TITLE
Initial Error email Api

### LINKED ISSUE #19: Replace apiPost errors with email or other
v1 - replace with next js api route, resend/mailgun/etc
v2 - add to database for querying/sorting

### PR DESCRIPTION
**Pull request by mdroidian**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new "Dev" tab in the Settings dialog for testing error email functionality.
	- Added a new component for rendering error report email templates.
	- New API route for handling error reporting via email.
	- Added a new type for error email properties.

- **Bug Fixes**
	- Enhanced error handling by implementing email notifications for errors instead of previous methods.

- **Documentation**
	- Updated package dependencies to support error reporting features.

- **Chores**
	- Updated configuration to improve environment variable handling during the build process.

### CODE REVIEW

**Comment by mdroidian** [comment]
This is a curious fail: https://vercel.com/discourse-graphs/discourse-graph/BWjiFXKzLuaXfnD4A2SJgXuDzyk6?filter=errors

```
This error was caused because patch-package cannot apply the following patch file:
patches/@tldraw+tldraw++@tldraw+editor+2.0.0-canary.ffda4cfb.patch
```

locally, seems to run fine.

❯ npm run postinstall

> roam@0.1.0 postinstall
> patch-package

patch-package 6.5.1
Applying patches...
@playwright/test@1.29.0 ✔
@tldraw/primitives@2.0.0-canary.ffda4cfb ✔
@tldraw/tldraw/@tldraw/editor@2.0.0-canary.ffda4cfb ✔
@tldraw/tldraw/@tldraw/ui@2.0.0-canary.ffda4cfb ✔
@tldraw/tlschema@2.0.0-canary.ffda4cfb ✔

File: apps/website/app/api/errors/route.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    return response;
+  } catch (error) {
+    return Response.json(
+      { error: error instanceof Error ? error.message : "Unknown error" },
+      { status: 500 },
+    );
+  }
+}
+
+// Add OPTIONS handler for preflight requests
```
Comment (sid597) [inline apps/website/app/api/errors/route.tsx]: Looks like its done?

**Comment by sid597** [comment]
LGTM, there are a few comments in apps/website/app/api/errors/route.tsx which seem redundant, its your call if you want to remove them.

**Review by sid597 — COMMENTED** [review]
LGTM, there are a few comments in apps/website/app/api/errors/route.tsx which seem redundant, its your call if you want to remove them.

**Pull request merged**